benchexec/tools/metaval.py

```diff
@@ -0,0 +1,89 @@
+"""
+BenchExec is a framework for reliable benchmarking.
+This file is part of BenchExec.
+
+Copyright (C) 2007-2019  Dirk Beyer
+All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import benchexec.util as util
+import benchexec.tools.template
+import benchexec.result as result
+import benchexec.tools.cpachecker as cpachecker
+import os
+
+
+class Tool(benchexec.tools.template.BaseTool):
+
+    REQUIRED_PATHS = ["CPAchecker", "esbmc"]
+    TOOL_TO_PATH_MAP = {"cpachecker": ["CPAchecker", "scripts"], "esbmc": ["esbmc"]}
+
+    def executable(self):
+        return util.find_executable("metaval.sh")
+
+    def name(self):
+        return "metaval"
+
+    def determine_result(self, returncode, returnsignal, output, isTimeout):
+        if not hasattr(self, "wrappedTool"):
+            return "METAVAL ERROR"
+        return self.wrappedTool.determine_result(
+            returncode, returnsignal, output, isTimeout
+        )
+
+    def cmdline(self, executable, options, tasks, propertyfile=None, rlimits={}):
+        verifierOption = options.index("--metaval")
+        self.verifierName = options[verifierOption + 1].lower()
+        del options[verifierOption : verifierOption + 2]
+        witnessOption = options.index("--witness")
+        self.witnessName = options[witnessOption + 1]
+        del options[witnessOption : witnessOption + 2]
+        if self.verifierName == "cpachecker":
```

Comment:
Why not do this in a generic way without per-verifier branching?

Using `import` will fail with tool-info modules that have a `-` in their name.

One could even reuse the import logic from BenchExec (`benchexec.model.load_tool_info`).

Reply:
I do not fully understand what you mean. I will have a look at  `benchexec.model.load_tool_info` once I have time.

Reply:
Using `benchexec.model.load_tool_info` does not work in all circumstances, because it is overwritten by the [Monkey-patch™](https://github.com/sosy-lab/cpachecker/blob/7bf51638177f593549e98fc1985752af12dd3903/scripts/benchmark.py#L192) in CPAcheckers `benchmark.py`.
I used the mechanism from `benchexec.model.load_tool_info` to instantiate the tool.